src/parse.ts

```diff
@@ -75,18 +75,18 @@ export const parser: Parser<BaseNode | undefined> = {
   ...typescript,
   astFormat: PRINTER_NAME,
 
-  preprocess(text: string, options: Options): string {
-    definePrinter(options);
+  // @ts-expect-error - preprocess hook can be async, but prettier types do not reflect that yet
+  async preprocess(text: string, options: Options): Promise<string> {
+    await definePrinter(options);
```

Comment:
I might have had a misunderstanding.

The v3 upgrade docs say that the _printer_ preprocess hook is async. But this is the _parser_ preprocess hook :cry:

Reply:
Seems like we need to use the `parse` method instead, apparently `preprocess` is on the chopping block for removal.

Is there anything I can do to help get this over the line @Techn1x ?

Reply:
Yeah I read that too. Unfortunately I think I might be out of my depth here, I took this on because I thought it was a straightforward upgrade but I think it needs deeper knowledge of prettier and this plugin than I have (which is not much, to be honest)

I don't know why the "hacks" exist that implement definePrinter() here for example

Very keen for someone else to carry the torch forward on this one